src/sgp/VSurface.h

```diff
@@ -50,8 +52,45 @@ class SGPVSurface
 		void ShadowRect(INT32 x1, INT32 y1, INT32 x2, INT32 y2);
 		void ShadowRectUsingLowPercentTable(INT32 x1, INT32 y1, INT32 x2, INT32 y2);
 
-		/* Allow read access to the underlying SDL_Surface */
-		SDL_Surface const& GetSDLSurface() const noexcept { return *surface_; }
+		/* Allow access to the underlying SDL_Surface */
+		SDL_Surface & GetSDLSurface() const noexcept { return *surface_; }
+
+		/* Return a software renderer for this surface. The renderer is created
+		 * on demand the first time this method gets called and will be
+		 * destroyed together with the surface. */
```

Comment:
I don't know how long-lived our SGPVSurfaces are or if SDL_CreateSoftwareRenderer is an expensive call (doesn't look like it from SDL source). Is it something we should worry about?

Reply:
99% of all uses of the software renderer will probably be on the frame buffer, which is created once at startup and lives until the program exits. I came to the same conclusion after looking at the SDL sources, a creating a software renderer seems to involve nothing more than filling out a couple of fields based on the surface's value. I just didn't want to create a renderer for every surface because most will never need one.

But now that we seem to have managed to motivate @maxrd2 to continue his work on #738, we may want to put this on hold. If I create even more merge conflicts for him he'll probably hit me. :laughing: